bigquery/google/cloud/bigquery/job.py

```diff
@@ -676,6 +693,10 @@ def from_api_repr(cls, resource):
         config.skip_leading_rows = _int_or_none(slr)
         if config.skip_leading_rows is None:
             del config.skip_leading_rows
+        if 'destinationEncryptionConfiguration' in resource:
+            key = resource['destinationEncryptionConfiguration']['kmsKeyName']
```

Comment:
`resource['destinationEncryptionConfiguration']['kmsKeyName']` could cause a problem if `resource['destinationEncryptionConfiguration']` is an empty dictionary. Use `resource['destinationEncryptionConfiguration'].get('kmsKeyName')`